lib/api/src/sys/module.rs

```diff
@@ -268,7 +268,7 @@ impl Module {
             let instance_handle = self.artifact.instantiate(
                 self.store.tunables(),
                 resolver,
-                Box::new((self.store.clone(), self.artifact.clone())),
+                Box::new((self.artifact.clone(), self.store.clone())),
```

Comment:
Actually it might make more sense to just replace this with:

```suggestion
                Box::new((self.clone(),))
```

That way we move the drop responsibility to the module itself. Can you try if this also fix your issue @YunSuk-Yeo ?

Reply:
okay, let me try.

Reply:
This one is working, so updated PR.